color/src/parse.rs

```diff
@@ -240,7 +240,7 @@ impl<'a> Parser<'a> {
     }
 
     fn raw_ch(&mut self, ch: u8) -> bool {
-        if self.s[self.ix..].as_bytes().first() == Some(&ch) {
+        if self.s.as_bytes().get(self.ix) == Some(&ch) {
```

Comment:
```suggestion
        debug_assert!(ch.is_ascii());
        if self.s.as_bytes().get(self.ix) == Some(&ch) {
```
Thoughts?

Reply:
The check is worth having, but is not necessary for equivalence (the only way these two differ is if `self.ix` is not at a char boundary, in which case the original panics).

Reply:
True; if the debug assert were that `.get(self.ix).is_ascii()`, that would be the equivalent